src/log.rs

```diff
@@ -17,3 +20,49 @@ pub const DEBUG: tracing::Level = tracing::Level::DEBUG;
 pub const INFO: tracing::Level = tracing::Level::INFO;
 #[cfg(feature = "tracing")]
 pub const WARN: tracing::Level = tracing::Level::WARN;
+
+///A record containing all of the logging metadata from librdkafka
+#[derive(Debug, Clone)]
+pub struct LogRecord {
+    level: RDKafkaLogLevel,
+    fac: String,
+    log_message: String,
+    contexts: String,
+}
+
+impl LogRecord {
+    /// Build a new LogRecord
+    pub fn new(level: RDKafkaLogLevel, fac: String, log_message: String, contexts: String) -> Self {
+        LogRecord {
+            level,
+            fac,
+            log_message,
+            contexts,
+        }
+    }
+
+    /// The librdkafka log level for this record
+    pub fn level(&self) -> RDKafkaLogLevel {
+        self.level
+    }
+
+    /// The librdkafka facility for this record
+    pub fn facility(&self) -> &str {
+        &self.fac
+    }
+
+    /// The message for this record
+    pub fn log_message(&self) -> &str {
+        &self.log_message
+    }
+
+    /// The set of debug contexts for this record
+    pub fn contexts(&self) -> &str {
+        &self.contexts
+    }
+
+    /// An iterator over the CSV context items
+    pub fn split_contexts(&self) -> Split<'_, &str> {
+        self.contexts.split(",")
+    }
```

Comment:
From the outside looking in, to me this stands out as a "raw form and ergonomic form" sort of interface. I don't use this code but I could imagine that if the contexts are delimited in some way, it's because the general intent is to be able to work with them individually.

All of that said, I might just rename `split_contexts` to `contexts` and `contexts` to `raw_contexts`, with a doc comment on (now renamed) `raw_contexts` explaining that the items are comma separated.

Reply:
Fixed in 68e09d1530269bfcaef55de370a6550627721876 .